elastic/kibana
# [License Management] Add URL locator

## Summary

Fixes https://github.com/elastic/kibana/issues/153104
Fixes https://github.com/elastic/kibana/issues/153037

This PR adds a URL locator to the License Management plugin that can be used by other plugins to safely render links to that plugin. An example of that is implemented in the Watcher plugin. 

### How to test
Since there is some logic in the code that disables Watcher if the license is not valid for that feature, it is actually not very probable to run into an issue with an invalid license in the UI. But we still can mock the license status and test the changes of this PR:
1. Start ES with the trial license `yarn es snapshot --license=trial` and start Kibana `yarn start`.
2. Update this [file](https://github.com/elastic/kibana/blob/main/x-pack/plugins/watcher/public/application/app.tsx) to mock an invalid license status: change line 61 to `if (true) {`. 
3. Navigate to Watcher to see that there is a link to License Management.
4. Update this [file](https://github.com/elastic/kibana/blob/main/x-pack/plugins/watcher/public/application/app.tsx) to mock an undefined License Management URL locator (imitates that the License Management plugin is disabled): change line 63 to `<LicensePrompt message={message} />`.
5. Navigate to Watcher to see that there is not link to License Management.

#### Screenshots
When the license status is invalid and the License Management plugin is enabled (not changed)
<img width="1259" alt="Screenshot 2023-03-28 at 14 55 54" src="https://user-images.githubusercontent.com/6585477/228833667-a96337cc-52c4-40fc-9209-37f7c0adab33.png">

When the license status is invalid and the License Management plugin is disabled
<img width="1258" alt="Screenshot 2023-03-28 at 14 55 27" src="https://user-images.githubusercontent.com/6585477/228834043-7aa270ba-92c4-48cd-9c75-3a060e3924eb.png">


### Checklist
